### PR TITLE
Implement two's complement encoding and decoding

### PR DESCRIPTION
**Pull request by axic**
This a rebased and polished version of https://github.com/indutny/bn.js/issues/73

### CODE REVIEW

File: test/utils-test.js [hunk 1]
```diff
@@ -207,4 +207,40 @@ describe('BN.js/Utils', function () {
...
+    it('should convert from two\'s complement to negative number', function () {
+      assert.equal(new BN('00000000', 16).fromTwos(32).toNumber(), 0);
+      assert.equal(new BN('00000001', 16).fromTwos(32).toNumber(), 1);
+      assert.equal(new BN('7fffffff', 16).fromTwos(32).toNumber(), 2147483647);
+      assert.equal(new BN('80000000', 16).fromTwos(32).toNumber(), -2147483648);
+      assert.equal(new BN('f0000000', 16).fromTwos(32).toNumber(), -268435456);
+      assert.equal(new BN('f1234567', 16).fromTwos(32).toNumber(), -249346713);
+      assert.equal(new BN('ffffffff', 16).fromTwos(32).toNumber(), -1);
+      assert.equal(new BN('fffffffe', 16).fromTwos(32).toNumber(), -2);
+      assert.equal(new BN('fffffffffffffffffffffffffffffffe', 16).fromTwos(128).toNumber(), -2);
```
Comment (indutny) [inline test/utils-test.js]: 80 column limit?

Comment (axic) [inline test/utils-test.js]: The tests pass - was that limit removed from the tests?

Comment (indutny) [inline test/utils-test.js]: Yeah, looks like semistandard doesn't have this check. Please fix it anyway.

Comment (axic) [inline test/utils-test.js]: I think its done with the last commit (also rebased to remove conflicts)

**Comment by indutny** [comment]
Yay, LGTM.

**Comment by indutny** [comment]
Thank you, landed and released.

File: test/utils-test.js (hunk 1 above)
Comment (dcousens) [inline test/utils-test.js]: @indutny 80 char limit?  What do I do with all my screen space?

**Comment by indutny** [comment]
@dcousens I'm not sure where is your comment, but this is what I do with my screen. 
screen shot 2016-01-24 at 10 05 15 pm

**Pull request closed without merging**